plugins/node/opentelemetry-instrumentation-pg/src/pg.ts

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  isWrapped,
+  InstrumentationBase,
+  InstrumentationConfig,
+  InstrumentationNodeModuleDefinition,
+} from '@opentelemetry/instrumentation';
+
+import {
+  context,
+  getSpan,
+  Span,
+  SpanKind,
+  StatusCode,
+} from '@opentelemetry/api';
+import * as pgTypes from 'pg';
+import * as pgPoolTypes from 'pg-pool';
+import {
+  PgClientExtended,
+  NormalizedQueryConfig,
+  PostgresCallback,
+  PgPoolExtended,
+  PgPoolCallback,
+} from './types';
+import * as utils from './utils';
+import { AttributeNames } from './enums';
+import { VERSION } from './version';
+
+export interface PostgresInstrumentationConfig {
```

Comment:
Should we agree on a name here? The instrumentation uses `pg` but here we use `Postgres`.

Reply:
please check the name of plugin

Reply:
I don't agree here as this is a new export in a new npm module. It was not named `InstrumentationConfig` in plugin.
Once it's released a name change is a breaking change. So if we want to get this aligned then in this PR or never.

Reply:
Previously we had `PostgresPlugin` now we have `PostgresInstrumentation`. `PostgresInstrumentationConfig` is a config for `PostgresInstrumentation`.  

`PostgresPlugin` don't have its own dedicated config, but `PostgresInstrumentation` has it.

Now by saying a breaking change and `InstrumentationConfig` what are you referring to as `InstrumentationConfig` is a config from instrumentation package ?.

Reply:
The npm module is named `@opentelemetry/instrumentation-pg` - so we use `pg` here. Looking into the type definitons of module pg the use als `pg` and `Pg`.

But our instrumentation uses `Postgres` as prefix for exported types - not `pg` or `Pg` which seems inconsistent to me.

Regarding breaking: Once we publish a version which exports something using `Postgres` as prefix it's a breaking change to rename it to use only `Pg` as prefix.

I have no strong opinion here regarding the prefix to use. Just want to note the the time to agree is now.

Reply:
It is currently like this already `opentelemetry-plugin-pg` and `PostgresPlugin` so I renamed the `plugin` with `instrumentation` like we do for any other plugins. For me the better naming would be postgres then pg as it better encapsulates what is this really about, but quite often the shortcut for postgres is simply `pg`, so `@opentelemetry/instrumentation-pg` is just shorter. But if I were to choose just one name I would go with `@opentelemetry/instrumentation-postgres` then

Reply:
My vote would be pg.
I like Postgres as name more as it is more descriptive but the NPM module instrumented uses pg and actually there is already a NPM module named postgres.

Let's wait what others have to say here :o)

Reply:
I think `Pg` because `pg` is the name of the module we are actually instrumenting and this follows the `CamelCasing` norm of js/ts.

Reply:
actually it is tricky because we are instrumenting `pg` and `pg-pool`

Reply:
anyway 2 votes for pg/Pg

Reply:
`pg-pool` is an internal module used by `pg`. For end users, they only ever see `pg`. Even if that wasn't the case, I think `pg` is closer to both than `postgres` is to either.

Reply:
I'm in favor of `PgInstrumentation`/`PgInstrumentationOptions` too

Reply:
Renamed Postgres to Pg